### ISSUE TITLE
JDK 9 Support

### ISSUE THREAD

**Issue by JLLeitschuh** [RuedigerMoeller/fast-serialization#219]
Does this work in JDK 9? Are the generated objects compatible with JDK 9 serialization?

**Comment by RuedigerMoeller** [comment]
I haven't tried yet. It probably won't work with modules out of the box due to new reflection limitations.

**Comment by JLLeitschuh** [comment]
Also, JDK 9 removed/moved `sun.misc.Unsafe` which means that it won't work.

**Comment by RuedigerMoeller** [comment]
That's right. It will be portable, but a probably specialized build will be necessary as AFAIK multi version builds will not work with jdk 8

**Comment by coldwarrl** [comment]
For my project it does not work with Java 9.

I get a lot of stuff like
`WARNING: Illegal reflective access by org.nustaq.serialization.FSTClazzInfo (file:/ivy2/cache/de.ruedigermoeller/fst/bundles/fst-2.56.jar) to field java.util.Random.seed
WARNING: Illegal reflective access by org.nustaq.serialization.FSTClazzInfo$FSTFieldInfo (file:/.ivy2/cache/de.ruedigermoeller/fst/bundles/fst-2.56.jar) to field java.util.Random.seed`

Is it planned that the library will support JDK 9 (by using modules, I guess) in the long run ?
Thanks.

**Comment by RuedigerMoeller** [comment]
unsafe is still there but has moved
https://dzone.com/articles/using-sunmiscunsafe-in-java-9-1
